src/dot.ts

```diff
@@ -5,30 +5,44 @@
 
 import { Vex } from './vex';
 import { Modifier } from './modifier';
+import { Note } from './note';
+import { StaveNote } from './stavenote';
 
 export class Dot extends Modifier {
-  static get CATEGORY() {
+  protected note?: Note;
+  protected radius: number;
+  protected dot_shiftY: number;
+
+  static get CATEGORY(): string {
     return 'dots';
   }
 
   // Arrange dots inside a ModifierContext.
-  static format(dots, state) {
+  static format(
+    dots: Dot[],
+    state: {
```

Comment:
I think this state is common to all the static formatter methods. Maybe it's a good idea to define an interface for this in `common`, perhaps `ModiferContextState`?

Reply:
It was already available in `ModifierContext`. :)
Changed.